parser/src/test/java/com/facebook/coresql/parser/sqllogictest/java/SqlLogicTest.java

```diff
@@ -0,0 +1,272 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.coresql.parser.sqllogictest.java;
+
+import com.facebook.coresql.parser.AstNode;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.FileVisitResult;
+import java.nio.file.FileVisitor;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.text.DecimalFormat;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static com.facebook.coresql.parser.ParserHelper.parseStatement;
+import static com.facebook.coresql.parser.Unparser.unparse;
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTestConfig.RESULTS_PATH;
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTestConfig.TEST_FILES_ROOT_PATH;
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTestUtils.createFileForResults;
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTestUtils.validateConfigData;
+import static java.lang.String.format;
+import static java.nio.file.FileVisitResult.CONTINUE;
+
+public final class SqlLogicTest
+{
+    private static SqlLogicTestConfig config;
+    private static Map<String, String> propertiesMap;
+    private static AcceptedRdms rdms;
+
+    private static Set<AcceptedRdms> skipIfSet;
+    private static AcceptedRdms onlyIf;
+    private static BufferedWriter writer;
+    private static FileVisitor<Path> fileVisitor;
+
+    private static int numFilesVisited;
+    private static int numStatementsParsed;
+    private static int numStatementsFound;
+
+    private enum SqlLogicTestError
+    {
+        PARSING_ERROR,
+        UNPARSING_ERROR,
+        UNPARSED_DOES_NOT_MATCH_ERROR
+    }
+
+    protected enum AcceptedRdms
+    {
+        ALL,
+        ORACLE,
+        MYSQL,
+        MSSQL,
+        SQLITE,
+        POSTGRESQL
+    }
```

Comment:
So what does this do? Can we add Presto or CoreSQL here? Or is it for validating results?

Reply:
Sqllogictest has optional flags that make dictate only certain RDMS should process certain queries. So, we need to track what RDMS we're currently testing against in order to get only queries relevant for that RDMS.

This enum contains all valid RDMS.

Done